File: 1827-minimum-operations-to-make-the-array-increasing/1827-minimum-operations-to-make-the-array-increasing.java
class Solution {
    public int minOperations(int[] nums) {
        int count = 0;
        int[] origin = Arrays.copyOf(nums, nums.length);
        int[] temp = new int[nums.length];
        temp[0] = nums[0];
        for (int i = 0; i < nums.length - 1; i++) {
            if (nums[i] >= nums[i + 1]) {
                nums[i + 1] = nums[i] + 1;
            }
            temp[i + 1] = nums[i + 1];
        }
        for (int i = 0; i < origin.length; i++) {
            count += (temp[i] - origin[i]);
        }
        return count;
    }
}

File: 1827-minimum-operations-to-make-the-array-increasing/1827-minimum-operations-to-make-the-array-increasing.kt
class Solution {
fun minOperations(nums: IntArray): Int {
    if (nums.size == 1) return 0

    var count = 0
    for (index in 0 until nums.size - 1) {
        if (nums[index] >= nums[index + 1]) {
            val diff = 1 + nums[index] - nums[index + 1]
            nums[index + 1] = nums[index] + 1
            count += diff
        }
    }

    return count
}
}